lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -136,22 +145,59 @@ class TraceElements extends Gatherer {
 
   /**
    * Find the node ids of elements which are animated using the Animation trace events.
+   * @param {LH.Gatherer.PassContext} passContext
    * @param {Array<LH.TraceEvent>} mainThreadEvents
-   * @return {Array<TraceElementData>}
+   * @return {Promise<Array<TraceElementData>>}
    */
-  static getAnimatedElements(mainThreadEvents) {
-    const animatedElementIds = new Set(mainThreadEvents
-      .filter(e => e.name === 'Animation' && e.ph === 'b')
-      .map(e => this.getNodeIDFromTraceEvent(e)));
+  static async getAnimatedElements(passContext, mainThreadEvents) {
+    const driver = passContext.driver;
+
+    /** @type Map<number, string[]> */
```

Comment:
should this be...
```suggestion
    /** @type Map<number, Set<string>> */
```

Reply:
Hmm I don't think we will ever encounter the same animation ID twice on one node. This is probably better to be safe.

Reply:
will we encounter them across different nodes/should it be a single large set for resolution?

just feels odd to have a structure about IDs if everything is a tree that will never be repeated, but I guess if that's the way the trace events work we can stick to it.

Reply:
The animation IDs should be unique across all nodes.